subworkflows/local/euk_mags_generation.nf

```diff
@@ -0,0 +1,260 @@
+    // ch_versions.mix( EUKCC_METABAT.out.versions.first() )
+
+    // -- prepare quality file
+    combine_quality = EUKCC_CONCOCT.out.eukcc_csv.join( EUKCC_METABAT.out.eukcc_csv )
+
+    // "genome,completeness,contamination" //
+    functionCATCSV = { item ->
+        def meta = item[0]
+        def list_files = [item[1], item[2]]
+        return tuple(meta, list_files)
+    }
+
+    CONCATENATE_QUALITY_FILES( combine_quality.map( functionCATCSV ), channel.value("quality_eukcc.csv") )
+
+    // ch_versions.mix( CONCATENATE_QUALITY_FILES.out.versions.first() )
+
+    quality = CONCATENATE_QUALITY_FILES.out.concatenated_result
+
+    // -- qs50 -- //
+    collect_data = quality.join( input.bins_concoct ) \
+        .join( input.bins_metabat ) \
+        .join( EUKCC_CONCOCT.out.eukcc_results ) \
+        .join( EUKCC_METABAT.out.eukcc_results )
+
+    FILTER_QS50( collect_data )
+
+    // ch_versions.mix( FILTER_QS50.out.versions.first() )
+
+    // input: tuple (meta, genomes/*, quality_file)
+    euk_drep_args = channel.value('-pa 0.80 -sa 0.99 -nc 0.40 -cm larger -comp 49 -con 21')
+
+    DREP( FILTER_QS50.out.qs50_filtered_genomes, euk_drep_args, channel.value('euk') )
+
+    // ch_versions.mix( DREP.out.versions.first() )
+
+    // -- aggregate by samples
+    // TODO: this collectFile is incorrect
+    quality_all_csv = quality.map { it -> it[1] }.collectFile(name: "all.csv", newLine: false)
+
+    MODIFY_QUALITY_FILE( quality_all_csv, channel.value("aggregated_euk_quality.csv"))
+
+    // ch_versions.mix( MODIFY_QUALITY_FILE.out.versions.first() )
+
+    aggregated_quality = MODIFY_QUALITY_FILE.out.modified_result.map { modified_csv ->
+        return tuple([id: "aggregated"], modified_csv)
+    }
+
+    // -- drep MAGs --//
+    euk_drep_args_mags = channel.value('-pa 0.80 -sa 0.95 -nc 0.40 -cm larger -comp 49 -con 21')
+
+    combine_drep = DREP.out.dereplicated_genomes.map{ it -> it[1] } \
+        .flatten() \
+        .collect() \
+        .map{ it ->
+            return tuple([id: "aggregated"], it)
+        }
+
+    DREP_MAGS( combine_drep.join( aggregated_quality ), euk_drep_args_mags, channel.value('euk_mags') )
+
+    // ch_versions.mix( DREP_MAGS.out.versions.first() )
+
+    // -- coverage -- //
+    bins_alignment = DREP.out.dereplicated_genomes.join( input.reads ) // tuple(meta, drep_genomes, [reads]),...
+
+    spreadBins = { record ->
+        if (record[1] instanceof List) {
+            return record
+        } else {
+            return tuple(record[0], [record[1]], record[2])
+        }
+    }
+
+    bins_alignment_by_bins = bins_alignment.map( spreadBins ).transpose(by: [1])  // tuple(meta, MAG1, [reads]); tuple(meta, MAG2, [reads])
+    
+    // TODO: leave only DREP_MAGS instead of DREP
+    ALIGN_BINS( bins_alignment_by_bins ) // out: [meta, fasta, bam, bai]
+
+    // ch_versions.mix( ALIGN_BINS.out.versions.first() )
+
+    // input: tuple(meta, MAG, bam, bai)
+    BREADTH_DEPTH( ALIGN_BINS.out.assembly_bam )
```

Comment:
We need to add a step to generate coverage per each MAG. MAG uploader needs file with 1 value per each MAG
example:
`SRR12830910_binner1.CONCOCT_v1.1.0-SRR12830910_bin.16.fa_coverage/SRR12830910_binner1.CONCOCT_v1.1.0-SRR12830910_bin.16.fa_MAGcoverage.txt`

```
SRR12830910_binner1.CONCOCT_v1.1.0-SRR12830910_bin.16.fa        0.7456465497737125
```